src/core/events/bpf.rs

```diff
@@ -152,44 +205,69 @@ impl EventFactory for BpfEventsFactory {
             0
         };
 
-        // Finally make our ring buffer and associate our map to our event
-        // processing closure.
-        let mut rb = libbpf_rs::RingBufferBuilder::new();
-        rb.add(&self.map, process_event)?;
-        let rb = rb.build()?;
-        let rs = self.run_state.clone();
-        // Start an event polling thread.
-        self.handle = Some(thread::spawn(move || {
-            while rs.running() {
-                if let Err(e) = rb.poll(Duration::from_millis(BPF_EVENTS_POLL_TIMEOUT_MS)) {
-                    match e.kind() {
-                        // Received EINTR while polling the
-                        // ringbuffer. This could normally be
-                        // triggered by an actual interruption
-                        // (signal) or artificially from the
-                        // callback. Exit without printing any error.
-                        libbpf_rs::ErrorKind::Interrupted => (),
-                        _ => error!("Unexpected error while polling ({e})"),
-                    }
-                    break;
+        let run_state = self.run_state.clone();
+        let process_log = move |data: &[u8]| -> i32 {
+            if data.len() != mem::size_of::<LogEvent>() {
+                error!("Unexpected log event size");
+                return 0;
+            }
+            // If a termination signal got received, return (EINTR)
+            // from the callback in order to trigger the event thread
+            // termination. This is useful in the case we're
+            // processing a huge number of buffers and rb.poll() never
+            // times out.
+            if !run_state.running() {
+                return -4;
+            }
+
+            let mut log_event = LogEvent::default();
+            if plain::copy_from_bytes(&mut log_event, data).is_err() {
+                error!("Can't read eBP log event");
+                return 0;
+            }
+
+            let log_level = match log_event.level {
+                1 => Level::Error,
+                2 => Level::Warn,
+                3 => Level::Info,
+                4 => Level::Debug,
+                5 => Level::Trace,
+                l => {
+                    error!("Unexpected log level ({l}). Falling back to error level");
+                    Level::Error
                 }
+            };
+
+            if let Ok(msg) = log_event.data.to_string() {
+                log!(log_level, "[eBPF] {msg}");
+            } else {
+                error!("Unable to convert eBPF log string");
```

Comment:
Ditto here. Using `match` can help getting the underlying error and show it in the log msg.

Reply:
this is a bit different from the previous one. Ok for `match` usage.